filter.php

```diff
@@ -55,11 +55,10 @@ public function filter($text, array $options = array()) {
             $renderer = $PAGE->get_renderer('filter_opencast');
 
             // Login if user is not logged in yet.
-            $loggedin = true;
-            if (!isset($_COOKIE['JSESSIONID']) && !self::$loginrendered) {
+            $loggedin = isset($_COOKIE['JSESSIONID']);
+            if (!$loggedin && !self::$loginrendered) {
                 // Login and set cookie.
                 filter_opencast_login();
-                $loggedin = false;
```

Comment:
This would only be reached for the first video rendered per page. For the following ones `$loggedin` would be `true`.

Reply:
That would lead to `$mustachedata->loggedin` being `false` only for the first video rendered per page. All following videos on that page are rendered with `loggedin` set to `true`. So the they would have the correct URL in their `src` attribute rather than in `data-frameSrc` and so the browser would load the videos before the LTI request is finished.